EGRESS-LOCKDOWN.md

```diff
@@ -0,0 +1,230 @@
+'marketplace.azurecr.io' \
+'marketplaceeush.cdn.azcr.io' \
+'ghcr.io' \
+'pkg-containers.githubusercontent.com' \
+'*.servicebus.windows.net')
+
+# Create the application rule for ACA Container Registry Access
+az network firewall application-rule create \
+-g $RG \
+-f $FIREWALLNAME \
+--collection-name 'aca-cr' \
+-n 'aca-cr' \
+--source-addresses '*' \
+--protocols 'http=80' 'https=443' \
+--target-fqdns ${TARGET_FQDNS[@]} \
+--action allow --priority 200
+
+
+az network firewall network-rule create \
+-g $RG \
+-f $FIREWALLNAME \
+--collection-name 'reddogfwnr' \
+-n 'svcbus' \
+--protocols 'TCP' \
+--source-addresses '*' \
+--destination-addresses  "ServiceBus" \
+--destination-ports 5671 443 \
+--action allow --priority 400
+
+az network firewall network-rule create \
+-g $RG \
+-f $FIREWALLNAME \
+--collection-name 'reddogfwnr' \
+-n 'sqltcp' \
+--protocols 'TCP' \
+--source-addresses '*' \
+--destination-addresses  "Sql" \
+--destination-ports 1433 11000-11999
+
+az network firewall network-rule create \
+-g $RG \
+-f $FIREWALLNAME \
+--collection-name 'reddogfwnr' \
+-n 'cosmos' \
+--protocols 'TCP' \
+--source-addresses '*' \
+--destination-addresses  "AzureCosmosDB" \
+--destination-ports 443 6380
+
+az network firewall network-rule create \
+-g $RG \
+-f $FIREWALLNAME \
+--collection-name 'reddogfwnr' \
+-n 'storage' \
+--protocols 'TCP' \
+--source-addresses '*' \
+--destination-addresses  "Storage" \
+--destination-ports 443
+
+az network firewall network-rule create \
+-g $RG \
+-f $FIREWALLNAME \
+--collection-name 'reddogfwnr' \
+-n 'monitor' \
+--protocols 'TCP' \
+--source-addresses '*' \
+--destination-addresses  "AzureMonitor" \
+--destination-ports 443
+```
+
+## Set up the Route Table
+
+```bash
+# Get the public and private IP of the firewall for the routing rules
+FWPUBLIC_IP=$(az network public-ip show -g $RG -n azfirewall-ip --query "ipAddress" -o tsv)
+FWPRIVATE_IP=$(az network firewall show -g $RG -n $FIREWALLNAME --query "ipConfigurations[0].privateIPAddress" -o tsv)
+
+# Create Route Table
+az network route-table create \
+-g $RG \
+-n acadefaultroutes
+
+# Create Default Routes
+az network route-table route create \
+-g $RG \
+--route-table-name acadefaultroutes \
+-n firewall-route \
+--address-prefix 0.0.0.0/0 \
+--next-hop-type VirtualAppliance \
+--next-hop-ip-address $FWPRIVATE_IP
+
+az network route-table route create \
+-g $RG \
+--route-table-name acadefaultroutes \
+-n internet-route \
+--address-prefix $FWPUBLIC_IP/32 \
+--next-hop-type Internet
+
+# Associate Route Table to ACA Subnet
+az network vnet subnet update \
+-g $RG \
+--vnet-name $VNET_NAME \
+-n aca \
+--route-table acadefaultroutes
+```
+
+## Deploy the Environment and Apps
+
+```bash
+ACARG=RedDogACAEgressLockdown-App
+LOC=eastus
+
+# Create the App Resource Group
+az group create -n $ACARG -l $LOC
+
+az deployment group create -n reddog -g $ACARG -f ./deploy/bicep/main.bicep --parameters vnetSubnetId=$PRIV_ACA_ENV_SUBNET_ID
```

Comment:
Cool. Yeah...I purposely switched off of consumption cuz I thought it would make the bicep template easier, and I wasnt sure if we typically showcase the consumption plan. I thought I'd need some conditional lines in there to ignore the workload profile section..but happy to give it a test with the workload profile name param. I'll test now and if it looks good I'll commit the update.

Thanks!

Reply:
Fixed in commit #36